app/controllers/vehicles_controller.rb

```diff
@@ -0,0 +1,50 @@
+class VehiclesController < ApplicationController
+  # GET /vehicles
+  def index
+    @vehicles = Vehicle.all
+
+    render json: @vehicles
+  end
+
+  # GET /vehicles/1
+  def show
+    render json: @vehicle
+  end
+
+  # POST /vehicles
+  def create
+    @vehicle = Vehicle.new(vehicle_params)
+
+    if @vehicle.save
+      render json: @vehicle, status: :created, location: @vehicle
+    else
+      render json: @vehicle.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /vehicles/1
+  def update
+    if @vehicle.update(vehicle_params)
+      render json: @vehicle
+    else
+      render json: @vehicle.errors, status: :unprocessable_entity
+    end
+  end
+
+  # DELETE /vehicles/1
+  def destroy
+    @vehicle.destroy
+  end
+
+  private
+
+  # Use callbacks to share common setup or constraints between actions.
+  def set_vehicle
+    @vehicle = Vehicle.find(params[:id])
+  end
+
+  # Only allow a list of trusted parameters through.
+  def vehicle_params
+    params.fetch(:vehicle, {})
+  end
+end
```

Comment:
Hello @degisew, you did not follow best practice for creating APIs by putting your controller in the app/controllers/api/v1 directory.